changelog.d/8501.bugfix

```diff
@@ -0,0 +1 @@
+Invalidate the cache when a fallback key is uploaded.
```

Comment:
This should match #8312 so they get grouped together during release:

```suggestion
Add support for olm fallback keys ([MSC2732](https://github.com/matrix-org/matrix-doc/pull/2732)).
```

Reply:
OK.  Does it need to be changed to `.feature` to match 8312 too?

Reply:
Oh, yep! Good call. 👍